packages/modules/imodel-browser/src/containers/iModelGrid/IModelGrid.tsx

```diff
@@ -109,7 +109,15 @@ export const IModelGrid = ({
   viewMode,
   maxCount,
 }: IModelGridProps) => {
-  const [sort, setSort] = useState<IModelSortOptions | undefined>(sortOptions);
+  const [sort, setSort] = React.useState<IModelSortOptions>(sortOptions);
+  const [isSortOnTable, setIsSortOnTable] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!isSortOnTable) {
```

Comment:
I would expect that if the grid props change, it would override the table props which would display the same thing, until they are changed manually in the header.

But for the external UI to actually be synchable, we should also now provide a callback onSortChange in the iModelGrid as well, so the external UI knows that the sort oder was changed within the component.

This was an oversight when we implemented the table view within the grid I'd say. Should always have been there since the moment we allowed the "grid" to update itself.

Reply:
I dont think the external UI should be in sync with the sorting we are doing on the table headers (if that's what you're implying). 

I see the sorting done on the table headers as being local changes to temporarily overwrite the sorting happening from the grid props. In the table headers, we can see the arrows showing that the sorting is being overridden.
![image](https://github.com/user-attachments/assets/0fc6ef26-1b67-4693-baee-b2d7fab84ce7)

If a user tries to change sorting through props while a sorting is applied on the table headers, it wouldn't be wrong to not affect the sorting since a more precise sorting is already being applied. Once the user removes the header sorting, they will then be able to see the props sorting that they were trying to do.
![image](https://github.com/user-attachments/assets/109606ef-c28c-4a5a-a69a-12b181db8190)

Reply:
Well, the sorting indicator sholdnt be seen as an override, this is a second bug that wasnt noticed; if the sorting of the  table is coming through the props, it should still display this indicator, as it clarifies to the user what sort is active (regardless of where it was set in the code)

Because right now we are creating the same issue we had in the first place, if you change the sort props, nothing will happen, as the table override this sorting.

What was the initial usecase that caused this issue from being raised? I would have expected that the UI shows the sorting options outside as the tile form of the iModelGrid does not allow selecting the sort option, which will likely look weird that the table is sorted in a way, while having a different element in the UI that explicitly sets the sorting to something else. As developpers we see the difference of who "override" what, but to a user it will only be a single option.

Reply:
But, as I'm no longer owning this component, if @aruniverse is fine with this feel free to resolve, I dont want to make this drag as I dont have enough time to keep an eye on the discussion with timely answers.

Reply:
The initial usecase was that in opensite+, we have a sort option outside of the grid, which as you've mentioned was added to sort the tile form but is still used for the table form as well.
![image](https://github.com/user-attachments/assets/179798ee-9484-44f5-b5e2-3797853f8ae2)

Reply:
I agree with Raphael, it makes sense when you are using tile view, but doesnt in the tabular view. Maybe completely disable external sort options when presenting the table form?